doc/develop/west/alias.rst

```diff
@@ -0,0 +1,56 @@
+.. _west-aliases:
+
+West aliases
+############
+
+West allows to add alias commands to the local, global or system configuration files.
+These aliases make it easy to add shortcuts for frequently used, or hard to memorize
+commands for ease of development.
+
+Similar to how ``git`` aliases work, the alias command is replaced with the alias'
+full text and parsed as a new shell argument list. This enables adding argument
+parameters as they were passed to the original command. Spaces are considered
+argument separators, use proper escaping if arguments shouldn't be split.
+
+To add a new alias simply call the ``west config`` command:
+
+.. code-block:: console
+
+   west config alias.command "other-command --with-arg=\"some value\""
+
```

Comment:
Add:
> To list aliases use `west help [some_alias`.

I noticed that `west -h some_alias` and `west help some_alias` differ. Not a problem but I hope this does not come as a surprise to you?

Reply:
Should be fixed in the west PR.

Reply:
not to mention `west -h <alias>` vs `west <alias> -h` which still gives different result.

Normally `west help <command>` == `west -h <command>` == `west <command> -h`.

Also I wonder if `west help` is the best command for showing all aliases.
`west config -l` will also list all aliases (and other configs), but `west help` lists not only aliases, but also all west built-in and extension commands.

For comparison, with a single alias command, then `west help` prints 59 lines of text, whereas `west config -l` prints just 5 lines.

And if I already know my alias I want to see, then a `west config alias.<some_alias>` is just as convenient :wink:

Reply:
> Not sure if this can/should be fixed.

As long as there is some way to get all the information and that each command output makes some kind of sense, I personally don't think "alias help consistency" matters much. Aliases are by definition a bit of "CLI hack"

Reply:
> not to mention west -h <alias> vs west <alias> -h which still gives different result.

Fixed in more recent versions.

I just approved the current version today BTW, I plan to merge it tomorrow if there's nothing new.